lib/neo4j/core/cypher_session/adaptors/faraday_helpers.rb

```diff
@@ -0,0 +1,49 @@
+module Neo4j
+  module Core
+    class CypherSession
+      module Adaptors
+        module FaradayHelpers
+          private
+
+          def verify_faraday_options(faraday_options = {}, defaults = {})
+            faraday_options.symbolize_keys!.reverse_merge!(defaults)
+            faraday_options[:adapter] ||= :net_http_persistent
+            require 'typhoeus/adapters/faraday' if faraday_options[:adapter].to_sym == :typhoeus
```

Comment:
Yes, that is true...  I was just bitten by https://github.com/typhoeus/typhoeus/issues/226#issuecomment-9919517 which was a little hard to track down the solution, so I figured it would be a little easier to just handle that for them instead (plus it is only needed here, not in their project generally).  Happy to take it out if you'd rather doc it / force folks to figure it out, but it seems like an easy win that doesn't really cost much here.

in the `Gemfile` I only include the adapter gems in :test (and not in the .gemspec) so tests here will fail if there's any issues vs letting people hit them when using whatever gems and fail and then report issues.

I can imagine if there *is* any issues with the adapter gem(s), it requires a little work to maintain the tests, but again, it seems like an easy thing we can stay on top of here vs just letting people get bitten and have to figure it out themselves..?

I'm of the understanding that only the `.gemspec` gems are exposed outwards, and so having gem in the `Gemfile` doesn't really impact anyone else - is that correct?